src/game/common/rts/team.cpp

```diff
@@ -1492,3 +1496,721 @@ bool Team::Damage_Team_Members(float amount)
+        obj = nullptr;
+    }
+
+    if (obj == nullptr) {
+        m_targetObjectID = OBJECT_UNK;
+    }
+
+    return obj;
+}
+
+void Team::Set_Override_Team_Relationship(unsigned int id, Relationship relationship)
+{
+    if (id != 0) {
+        m_teamRelations->m_relationships[id] = relationship;
+    }
+}
+
+void Team::Set_Override_Player_Relationship(int id, Relationship relationship)
+{
+    if (id != -1) {
+        m_playerRelations->m_relationships[id] = relationship;
+    }
+}
+
+bool Team::Remove_Override_Player_Relationship(int id)
+{
+    if (m_playerRelations->m_relationships.empty()) {
+        return false;
+    }
+
+    if (id == -1) {
+        m_playerRelations->m_relationships.clear();
+        return true;
+    }
+
+    auto it = m_playerRelations->m_relationships.find(id);
+
+    if (!(it != m_playerRelations->m_relationships.end())) {
+        return false;
+    }
+
+    m_playerRelations->m_relationships.erase(it);
+    return true;
+}
+
+bool Team::Is_Idle() const
+{
+    DLINK_ITERATOR<Object> it = Iterate_Team_Member_List();
+
+    while (!it.Done()) {
+        AIUpdateInterface *update = it.Cur()->Get_AI_Update_Interface();
+
+        if (update != nullptr) {
+            if (!it.Cur()->Is_Effectively_Dead() && !update->Is_Idle()) {
+                return false;
+            }
+        }
+
+        it.Advance();
+    }
+
+    return true;
+}
+
+void Team::Notify_Team_Of_Object_Death()
+{
+    const TeamTemplateInfo *info = m_proto->Get_Template_Info();
+
+    if (info != nullptr) {
+        if (!info->m_scriptOnUnitDestroyed.Is_Empty()) {
+            g_theScriptEngine->Run_Script(info->m_scriptOnUnitDestroyed, this);
+        }
+    }
+}
+
+bool Loco_Set_Matches(int lstm, unsigned int surface_bit_flags)
+{
+    return static_cast<unsigned char>((lstm & ((4 * (surface_bit_flags & 2)) | surface_bit_flags & 1))) != 0;
+}
+
+bool Team::Did_All_Enter(PolygonTrigger *poly, unsigned int surfaces) const
+{
+    if (!m_enteredOrExited) {
+        return false;
+    }
+
+    bool b1 = false;
+    bool b2 = false;
+    bool b3 = false;
+    DLINK_ITERATOR<Object> it = Iterate_Team_Member_List();
+
+    while (!it.Done()) {
+        AIUpdateInterface *update = it.Cur()->Get_AI_Update_Interface();
+
+        if (update != nullptr) {
+            if (!Loco_Set_Matches(update->Get_Locomotor_Set()->Get_Valid_Surfaces(), surfaces)) {
+                it.Advance();
+                continue;
+            }
+        } else if (!Loco_Set_Matches(1, surfaces)) {
+            it.Advance();
+            continue;
+        }
+
+        if (!it.Cur()->Is_Effectively_Dead()) {
+            if (!it.Cur()->Is_KindOf(KINDOF_INERT)) {
+                if (it.Cur()->Did_Enter(poly)) {
+                    b2 = true;
+                } else {
+                    if (!it.Cur()->Is_Inside(poly)) {
+                        b3 = true;
+                    }
+                }
+
+                b1 = true;
```

Comment:
`b1` is set, but never used for anything in this function. What is this for? Perhaps is missing from the return statement below?

Reply:
I don't know, its definitely how the original code looks.

Reply:
Compiler should have optimized it away then, if not used. Does b1 only show in World Builder only? Something is not right here.